src/components/Builder/Index.jsx

```diff
@@ -1,56 +1,77 @@
-import React, { useRef, useState } from "react";
+import React, { useEffect,useRef, useState } from "react";
+import { useParams } from "react-router-dom";
 import { Col, Radio, Row, Space, Switch, Tabs, Typography } from "antd";
+import { useGetBasicInfoQuery } from "../../api/profileApi";
 import { PROFILES } from "../../Constants";
 import Navbar from "../Navbar/navbar";
 import Resume from "../Resume/Resume";
+import Achievement from "./Achievement";
 import BasicInfo from "./BasicInfo";
 import styles from "./Builder.module.css";
 import Certification from "./Certification";
 import Education from "./Education";
 import Experience from "./Experience";
 import Project from "./Project";
-import Skills from "./Skills";
 
-const defaultPanes = [
+const createPanes = (profileData, disableTabs) => [
   {
     key: "basic-info",
     label: <b>Basic Info</b>,
-    children: <BasicInfo />
+    children: <BasicInfo profileData={profileData} />,
   },
   {
     key: "projects",
     label: <b>Projects</b>,
-    children: <Project />
+    children: <Project />,
+    disabled: disableTabs,
   },
   {
     key: "education",
     label: <b>Education</b>,
-    children: <Education />
+    children: <Education />,
+    disabled: disableTabs,
   },
   {
-    key: "skills",
-    label: <b>Skills</b>,
-    children: <Skills />
-  }
+    key: "experience",
+    label: <b>Experience</b>,
+    children: <Experience />,
+    disabled: disableTabs,
+  },
 ];
 
-const experience = {
-  key: "experience",
-  label: <b>Experience</b>,
-  children: <Experience />
-};
+const achievement = (profileData, disableTabs) => ({
+  key: "achievement",
+  label: <b>Achievement</b>,
+  children: <Achievement profileData={profileData} />,
+  disabled: disableTabs,
+});
 
-const certification = {
+const certification = (profileData, disableTabs) => ({
   key: "certification",
   label: <b>Certification</b>,
-  children: <Certification />
-};
+  children: <Certification profileData={profileData} />,
+  disabled: disableTabs,
+});
 
 export const Editor = () => {
-  const [items, setItems] = useState(defaultPanes);
+  const { id } = useParams();
+  const [items, setItems] = useState(createPanes(null, !id));
   const [profile, setProfile] = useState(PROFILES.internal);
-
   const resumeRef = useRef();
+  const [profileData, setProfileData] = useState(null);
+
+  const {data} = useGetBasicInfoQuery(id);
+
+  useEffect(() => {
+    if (id) {
+      if(data){
+        setProfileData(data);
+        setItems(createPanes(data, false));
+      }
+    } else {
```

Comment:
do we  need of else condition here ? if not remove it

Reply:
Done!